ReactWindows/ReactNative.Shared/ReactInstanceManager.cs

```diff
@@ -607,14 +609,23 @@ private void InvokeDefaultOnBackPressed()
             reactInstance.GetJavaScriptModule<AppRegistry>().runApplication(jsAppModuleName, appParameters);
         }
 
-        private void DetachViewFromInstance(ReactRootView rootView, IReactInstance reactInstance)
+        private async Task DetachViewFromInstanceAsync(ReactRootView rootView, IReactInstance reactInstance)
         {
             DispatcherHelpers.AssertOnDispatcher();
 
             var uiManagerModule = reactInstance.GetNativeModule<UIManagerModule>();
-            uiManagerModule.DetachRootView(rootView);
+
+            // Detaches ReactRootView from instance manager root view list and size change monitoring.
+            // Has to complete before unmounting application
+            await uiManagerModule.DetachRootViewAsync(rootView);
+
+            // Initiates a clean up operation that will wait for both a "removeRootView" command from JS (an effect of
+            // unmounting the application) and the release of all dispatcher affined UI objects.
+            var crTask = uiManagerModule.GetRootViewCleanupTask(rootView.GetTag());
```

Comment:
>.GetTag() [](start = 72, length = 9)

Is the root view guaranteed to have a tag? We ran into a problem recently where `GetTag()` was being called before the tag had been set.

Reply:
It is, the tag is created in the other half of the pair (AttachMeasuredRootViewToInstance)

---
In reply to: [178548427](https://github.com/Microsoft/react-native-windows/pull/1737#discussion_r178548427) [](ancestors = 178548427)